tests/ttnn/unit_tests/operations/test_moreh_dot.py

```diff
@@ -16,84 +16,59 @@
+def run_moreh_dot_test(input_shape, ttnn_dtype, device, optional_output=False):
+    # TODO @thanhnguyen-moreh: Support bfloat8_b in kernel
+    if ttnn_dtype == ttnn.bfloat8_b:
+        pytest.skip(f"bfloat8_b is not supported in the kernel")
+    torch_dtype = get_torch_dtype(ttnn_dtype)
+    output_shape = [1, 1, 1, 1]
 
-    # create tensors for forward
-    if use_randint:
-        input = torch.randint(-2, 3, input_shape, dtype=torch.bfloat16)
-        other = torch.randint(-2, 3, other_shape, dtype=torch.bfloat16)
-        output = torch.randint(-2, 3, output_shape, dtype=torch.bfloat16)
-    else:
-        input = torch.rand(input_shape, dtype=torch.bfloat16)
-        other = torch.rand(other_shape, dtype=torch.bfloat16)
-        output = torch.rand(output_shape, dtype=torch.bfloat16)
+    if ttnn_dtype == ttnn.int32:
+        torch_input = torch.randint(-2, 3, input_shape, dtype=torch_dtype)
+        torch_other = torch.randint(-2, 3, input_shape, dtype=torch_dtype)
 
-    # inputs must be of type bfloat16 or bfloat8
-    tt_input = create_tt_tensor(input, cpu_layout, ttnn.bfloat16).pad_to_tile(float(1)).to(npu_layout).to(device)
-    tt_other = create_tt_tensor(other, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
-    tt_output = create_tt_tensor(output, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
+        tt_input = ttnn.from_torch(
+            torch_input, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT
+        )  # Input dtype must be of bfloat16 or bfloat8_b
+        tt_other = ttnn.from_torch(
+            torch_other, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT
+        )  # Input dtype must be of bfloat16 or bfloat8_b
+    else:
+        torch_input = torch.rand(input_shape, dtype=torch_dtype)
+        torch_other = torch.rand(input_shape, dtype=torch_dtype)
 
-    torch_input = input.reshape(-1) if is_1d else input
-    torch_other = other.reshape(-1) if is_1d else other
+        tt_input = ttnn.from_torch(torch_input, device=device, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT)
+        tt_other = ttnn.from_torch(torch_other, device=device, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT)
 
-    # tensors for backward
-    output_grad = tt_output_grad = torch_output_grad = tt_input_grad = tt_other_grad = None
-    if require_input_grad or require_other_grad:
-        output_grad = torch.randint(-2, 3, output_shape, dtype=cpu_dtype)
-        tt_output_grad = (
-            create_tt_tensor(output_grad, cpu_layout, npu_dtype).pad_to_tile(float(-1)).to(npu_layout).to(device)
+    if optional_output == False:
+        tt_out = ttnn.operations.moreh.dot(tt_input, tt_other, dtype=ttnn_dtype)
+    else:
+        torch_optional_output = torch.rand(output_shape, dtype=torch_dtype)
+        optional_output = ttnn.from_torch(
+            torch_optional_output, device=device, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT
         )
```

Comment:
```suggestion
    if optional_output == False:
        tt_out = ttnn.operations.moreh.dot(tt_input, tt_other, dtype=ttnn_dtype)
    else:
        torch_optional_output = torch.rand(output_shape, dtype=torch_dtype)
        optional_output = ttnn.from_torch(
            torch_optional_output, device=device, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT
        )
```

Change to:
```python
optional_output = ttnn.from_torch(torch_optional_output, device=device, dtype=ttnn_dtype, layout=ttnn.TILE_LAYOUT) if use_optional_output else None
```

Same for other files

Reply:
Resolved